docs/architecture.md

```diff
@@ -0,0 +1,185 @@
+
+**Output:** `Network.operator.openshift.io/v1 cluster` ([typedef](https://github.com/openshift/api/blob/master/operator/v1/types_network.go))
+
+The Cluster Config controller reads the high-level configuration object, and applies it "downward" to the detailed network configuration (the poorly-named _Operator Configuration_). Any conflicting fields in the operator configuration will be overwritten.
+
+It partially generates the status of the `Network.config` object, but this object is a bit of an exception.
+
+## Operator Config Controller (Network Controller)
+
+**Input:** `Network.operator.openshift.io/v1 cluster` ([typedef](https://github.com/openshift/api/blob/master/operator/v1/types_network.go))
+
+**Output:** Most core networking components.
+
+This is the "main" controller, in that it is responsible for rendering the core networking components (openshift-sdn / OVNKubernetes / Kuryr, Multus, etc). It is broken down in to stages:
+
+1. **Validate** the supplied configuration
+2. **Fill** - Determine any unsupplied default values.
+3. **Check** - Compare against previously-applied configuration, to see if any unsafe changes are proposed
+4. **Bootstrap** - gather existing cluster state, and create any non-Kubernetes resources (i.e. OpenStack objects)
+5. **Render** - process template files in `/bindata` and generate Kubernetes objects
+6. **Apply** - Create or update objects in the APIServer. Delete any un-rendered objects.
+
+### Applied configuration
+
+The Network operator needs to make sure that the input configuration doesn't change unsafely, since we don't support rolling out most changes. To do that, it writes a ConfigMap with the applied changes. It then compares the existing configuration with the desired configuration, and sets a status of `Degraded` if it is asked to do something unsafe.
+
+The persisted configuration must **make all defaults explicit**. This protects against inadvertent code changes that could destabilize an existing cluster.
+
+## Egress Router
+
+**Input:** `EgressRouter.network.operator.openshift.io`
+**Output:** EgressRouters (a Deployment and a NetworkAttachmentDefinition)
+
+The EgressRouter is a feature that spins up a container with a MacVLAN secondary interface. Other containers can then NAT their traffic through that interface. The routing is handled via OVN-Kubernetes, but there needs to be a container to hold the interface. This controller watches for EgressRouter CRs and creates / deletes pods as required.
+
+## Ingress Config
+
+**Input:** `IngressController.operator.openshift.io`
+**Output:** A label
+
+The Ingress Config controller ensures that end-users can grant access to IngressControllers, even when they are located in host-network pods. If the IngressController reports host-network pods, then the controller will add the label `policy-group.network.openshift.io/ingress=""` to the special host-network holding namespace.
```

Comment:
This feels like it should have a link to the enhancement or something, because it's pretty unclear.

Actually, each of the "minor" operators could use a link to its corresponding feature documentation or something

Reply:
added docs to where I could find them.